client/allocrunner/taskrunner/stats_hook_test.go

```diff
@@ -150,10 +159,19 @@ func TestTaskRunner_StatsHook_Periodic(t *testing.T) {
 	require.NoError(h.Exited(context.Background(), nil, nil))
 
 	// Should *not* get another update in ~500ms (see interval above)
+	// we may get a single update due to race with exit
+	timeout := time.After(2 * interval)
+	firstUpdate := true
+
+WAITING:
 	select {
 	case ru := <-su.Ch:
+		if firstUpdate {
+			firstUpdate = false
+			goto WAITING
+		}
```

Comment:
Here, we also can be a bit more discerning to ensure that no stats update happens after the task is marked as exited, but It's not fully clear to me how we can completely rule out the race between exiting and channel sending; so I decided to ignore the first update after the exit as spurious update.